charging_stations_pipelines/pipelines/osm/osm.py

```diff
@@ -1,51 +1,119 @@
+    osm_mapper,
+    SCOPE_COUNTRIES,
+)
 from charging_stations_pipelines.pipelines.osm.osm_receiver import get_osm_data
-from charging_stations_pipelines.pipelines.station_table_updater import StationTableUpdater
+from charging_stations_pipelines.pipelines.station_table_updater import (
+    StationTableUpdater,
+)
+from charging_stations_pipelines.shared import JSON
 
 logger = logging.getLogger(__name__)
 
 
-class OsmPipeline:
-    def __init__(self, country_code: str, config: configparser, session: Session, online: bool = False):
+class OsmPipeline(Pipeline):
+    """Pipeline for the OSM data source."""
+
+    def __init__(
+        self,
+        country_code: str,
+        config: configparser,
+        session: Session,
+        online: bool = False,
+    ):
+        super().__init__(config, session, online)
+
         self.country_code = country_code
-        self.config = config
-        self.session = session
-        self.online: bool = online
-        self.data: Optional[Dict] = None
 
-    def _retrieve_data(self):
+    def retrieve_data(self):
         data_dir: str = os.path.join(
             pathlib.Path(__file__).parent.resolve(), "../../..", "data"
         )
         pathlib.Path(data_dir).mkdir(parents=True, exist_ok=True)
-        tmp_file_path = os.path.join(data_dir, self.config[DATA_SOURCE_KEY]["filename"])
+        tmp_file_path = os.path.join(
+            data_dir, self.config[osm.DATA_SOURCE_KEY]["filename"]
+        )
         if self.online:
             logger.info("Retrieving Online Data")
             get_osm_data(self.country_code, tmp_file_path)
         with open(tmp_file_path) as f:
             self.data = json.load(f)
 
     def run(self):
-        logger.info(f"Running {self.country_code} {DATA_SOURCE_KEY} Pipeline...")
-        self._retrieve_data()
+        logger.info(f"Running {self.country_code} {osm.DATA_SOURCE_KEY} Pipeline...")
+        self.retrieve_data()
         station_updater = StationTableUpdater(session=self.session, logger=logger)
-        data = self.data.get("elements", [])
-        for entry in tqdm(iterable=iter(data), total=len(data)):  # type: Dict
-            mapped_address = map_address_osm(entry, None)
-            mapped_charging = map_charging_osm(entry, None)
-            mapped_station = map_station_osm(entry, self.country_code)
-            mapped_station.address = mapped_address
-            mapped_station.charging = mapped_charging
-            station_updater.update_station(mapped_station, DATA_SOURCE_KEY)
+
+        stats = collections.defaultdict(int)
+        data_elements: JSON = self.data.get("elements", [])
+        entry: JSON
+        for entry in tqdm(iterable=iter(data_elements), total=len(data_elements)):
+            try:
+                station = osm_mapper.map_station_osm(entry)
+                # Filter out stations with country codes that are not in the scope of the pipeline
+                if station.country_code not in SCOPE_COUNTRIES:
```

Comment:
I'm not confident that we actually should filter the stations that have inconsistent data regarding user-entered info like country, address, etc. because this will often be inconsistent for the open street map data source.
The info that there is a station at this location is still valuable for the result (i.e. to be considered in the merger).
I think it is still valuable to show some statistics or details about these inconsistencies, especially in debug log level

Reply:
@mjmader agree, I remove filtering and left the counts. I.e. I just logging the counts and not removing anymore.

Reply:
@mjmader agree, I remove filtering and left the counts. I.e. I'm just logging the counts and not removing anything.